bpf-gpl/tc.c

```diff
@@ -455,6 +455,25 @@ static CALI_BPF_INLINE int calico_tc(struct __sk_buff *skb)
 		}
 	}
 
+	// Drop malformed IP packets
+	if (ip_header->ihl < 5) {
+		fwd.reason = CALI_REASON_IP_MALFORMED;
+		CALI_DEBUG("Drop malformed IP packets\n");
+		goto deny;
+	}
+	else if (ip_header->ihl > 5) {
+		/* Drop packets with IP options from/to WEP.
+		 * Also drop packets with IP options if the dest IP is not host IP
+		 */
+		if (CALI_F_FROM_WEP || CALI_F_TO_WEP ||
+			(CALI_F_FROM_HEP && !cali_rt_flags_local_host(cali_rt_lookup_flags(ip_header->daddr)))) {
```

Comment:
I have introduced `rt_addr_is_local_host()` for another usecase in https://github.com/projectcalico/felix/pull/2326